homeassistant/components/overkiz/climate_entities/somfy_thermostat.py

```diff
@@ -0,0 +1,178 @@
+    ) -> None:
+        """Init method."""
+        super().__init__(device_url, coordinator)
+        self.temperature_device = self.executor.linked_device(2)
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return hvac operation ie. heat, cool mode."""
+        return OVERKIZ_TO_HVAC_MODES[
+            self.executor.select_state(CORE_DEROGATION_ACTIVATION_STATE)
+        ]
+
+    @property
+    def hvac_action(self) -> str:
+        """Return the current running hvac operation if supported."""
+        if not self.current_temperature or not self.target_temperature:
+            return CURRENT_HVAC_IDLE
+        if self.current_temperature < self.target_temperature:
+            return CURRENT_HVAC_HEAT
+        return CURRENT_HVAC_IDLE
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode, e.g., home, away, temp."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            return OVERKIZ_TO_PRESET_MODES[
+                self.executor.select_state(ST_HEATING_MODE_STATE)
+            ]
+        return OVERKIZ_TO_PRESET_MODES[
+            self.executor.select_state(ST_DEROGATION_HEATING_MODE_STATE)
+        ]
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return cast(
+            float,
+            self.temperature_device.states.get(OverkizState.CORE_TEMPERATURE).value,
+        )
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        if self.hvac_mode == HVAC_MODE_AUTO:
+            if self.preset_mode == PRESET_NONE:
+                return None
+            return self.executor.select_state(TARGET_TEMP_TO_OVERKIZ[self.preset_mode])
+        return self.executor.select_state(CORE_DEROGATED_TARGET_TEMPERATURE_STATE)
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temperature = kwargs[ATTR_TEMPERATURE]
+
+        await self.executor.async_execute_command(
+            COMMAND_SET_DEROGATION, temperature, STATE_DEROGATION_FURTHER_NOTICE
+        )
+        await self.executor.async_execute_command(
+            COMMAND_SET_MODE_TEMPERATURE, STATE_PRESET_MANUAL, temperature
+        )
+        await self.executor.async_execute_command(COMMAND_REFRESH_STATE)
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target hvac mode."""
+        if hvac_mode == HVAC_MODE_AUTO:
+            await self.executor.async_execute_command(COMMAND_EXIT_DEROGATION)
+            await self.executor.async_execute_command(COMMAND_REFRESH_STATE)
+        else:
+            await self.async_set_preset_mode(PRESET_NONE)
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set new preset mode."""
+        if self.preset_mode == preset_mode:
+            return
```

Comment:
Are we sure that this can happen? I don't think so. We don't add this extra check in other climate implementations.

Reply:
Indeed, I forgot to remove this one.